docs/handbook/engineering/enterprise-edition.md

```diff
@@ -0,0 +1,111 @@
+
+#### Helpers in ruby Code
+While working in the ruby code, `ChatwootApp.enterprise?` helper is available for you to determine whether the user is running the Enterprise Edition of the software.
+
+#### Data Models & Data Migrations
+Users could be migrating between versions. So we always have to ensure that all the migrations are run for both the edition. The database schema needs to remain the same for both editions. In the Case of DataMigrations, you can have blank implementations for these migrations by conditionally excluding Enterprise specific DataModels with 
+`ChatwootApp.enterprise?`
+
+### Routes
+In cases where you want to limit the the access of a specific route to `Enterprise` Edition alone, use the `ChatwootApp.enterprise?` helper.
+
+```
+  if ChatwootApp.enterprise?
+    get '/enterprise', to: 'dashboard#enterprise'
+  else
+```
+
+### Developing Features as an extension of Community Edition
+
+When developing Enterprise features built over the Community Edition software, Implement a base functionality in Community edition code and extend it over in an Enterprise licensed module.
+
+As shown in the example, we do this by writing a module in the Enterprise namespace and injecting it into the community edition class.
+
+```
+# app/models/inbox.rb
+class Inbox < ApplicationRecord
+  # placeholder method which will be overriden in Enterprise edition
+  def member_ids_with_assignment_capacity
+    members.ids
+  end
+end
+
+Inbox.prepend_mod_with('Inbox')
+```
+
+```
+# enterprise/app/models/enterprise/inbox.rb
+module Enterprise::Inbox
+  def member_ids_with_assignment_capacity
+    super - get_agent_ids_over_assignment_limit
+  end
+
+  private
+
+  def get_agent_ids_over_assignment_limit
+    # implement proprietry enterprise logic
+  end
+end
+
+```
+You can use the helper methods like `prepend_mod_with`, `extend_mod_with`, or `include_mod_with`, depending on how you want to include the enterprise method into the chain. Refer [InjectEnterpriseEditionModule](https://github.com/chatwoot/chatwoot/blob/develop/config/initializers/01_inject_enterprise_edition_module.rb) to learn more.
+
+Ensure that the module written to extend the Community Logic has to be put under the `Enterprise` namespace.
+```
+# notice the extra namespacing with in models directory
+# enterprise/app/models/enterprise/inbox.rb
+
+# other examples
+# enterprise/app/controllers/enterprise/paid_feature_controller.rb
+# enterprise/app/builders/enterprise/paid_builder.rb
+
+```
+
+
+### Developing Feature Exclusive for Enterprise Edition
+
+If the feature is not present in Community Edition, We must directly put the code into the enterprise directory without the enterprise namespace. This works because we have the enterprise directory in [autoload paths](https://github.com/chatwoot/chatwoot/blob/41b89014324772d8351b15936d25623f852980cb/config/application.rb#L19).
```

Comment:
```suggestion
If the feature is not present in Community Edition, we must directly put the code into the enterprise directory without the enterprise namespace. This works because we have the enterprise directory in [autoload paths](https://github.com/chatwoot/chatwoot/blob/41b89014324772d8351b15936d25623f852980cb/config/application.rb#L19).
```